### PR TITLE
implement pub/sub model

### PR DESCRIPTION
**Pull request by chhsiao90**
solve #13  ~~

### CODE REVIEW

File: proxy.py [hunk 1]
```diff
@@ -1,28 +1,21 @@
...
 import tornado.iostream
 import tornado.netutil
 import tornado.gen
 
+import logging
+
 import http
-from http import HttpMessage, HttpMessageBuilder
+from http import HttpMessageBuilder
+from utils import curr_loop
```
Comment (mike820324) [inline proxy.py:13]: Missing utils.py.

**Comment by mike820324** [comment]
It seems like the zmq socket of msg_publisher did not use tornado iostream. 
Any particular reason why not using iostream?

Will the whole ioloop been block when zmq is publishing the msg?

File: msg_publisher.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class MsgPublisher(object):
+    def __init__(self, zmq_socket):
+        self.zmq_socket = zmq_socket
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logging.DEBUG)
+
+    def publish(self, message):
+        self.zmq_socket.send_json(message)
```
Comment (mike820324) [inline msg_publisher.py:14]: Using socket instead of zmq tornado iostream, will this part being blocked?

Comment (mike820324) [inline msg_publisher.py:14]: Okay, I might know why you dis not using the tornado ioloop. Is it the callback?

Comment (chhsiao90) [inline msg_publisher.py:14]: the zmq socket is non-blocking, just like iostream work.
We keep iostream, and call it's send method when we need.

File: proxy.py (hunk 1 above)
Comment (chhsiao90) [inline proxy.py:13]: Haha, sorry, my mistake.
I had commit utils.py!

**Pull request merged**